java/serve/src/main/java/io/ray/serve/handle/DeploymentHandle.java

```diff
@@ -0,0 +1,128 @@
+package io.ray.serve.handle;
+
+import io.ray.runtime.metric.Count;
+import io.ray.runtime.metric.Metrics;
+import io.ray.serve.api.Serve;
+import io.ray.serve.common.Constants;
+import io.ray.serve.deployment.DeploymentId;
+import io.ray.serve.generated.RequestMetadata;
+import io.ray.serve.metrics.RayServeMetrics;
+import io.ray.serve.router.Router;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+
+/** A handle to a service deployment. */
+public class DeploymentHandle implements Serializable {
+
+  private static final long serialVersionUID = 550701184372753496L;
+
+  private DeploymentId deploymentId;
+
+  private HandleOptions handleOptions;
+
+  private String handleTag;
+
+  private transient Count requestCounter;
+
+  private transient volatile Router router;
+
+  public DeploymentHandle(
+      String deploymentName, String appName, HandleOptions handleOptions, Router router) {
```

Comment:
make router as optional.

Reply:
Added overloaded constructor for DeploymentHandle, allowing the router to be omitted during construction.